src/device-api/common.coffee

```diff
@@ -38,6 +39,13 @@ exports.doPurge = (applications, appId, force) ->
 			currentState.local.apps[appId] = purgedApp
 			deviceState.pausingApply ->
 				deviceState.applyIntermediateTarget(currentState, { skipLock: true })
+				.then ->
+					# Now that we're not running anything, explicitly
+					# remove the volumes, we must do this here, as the
+					# application-manager will not remove any volumes
+					# which are part of an active application
+					volumes.getAllByAppId(appId).then (vols) ->
+						Bluebird.each(vols, (vol) -> vol.remove())
```

Comment:
This could be
```coffeescript
Bluebird.each(volumes.getAllByAppId(appId), (v) -> v.remove())
```

Reply:
Ah yeah nice, let bluebird do the resolution, good catch :+1: